sklearn/model_selection/_classification_threshold.py

```diff
@@ -0,0 +1,895 @@
+        # add `pos_label` if requested by the scorer function
+        scorer_kwargs = {**scorer._kwargs}
+        signature_scoring_func = signature(scorer._score_func)
+        if (
+            "pos_label" in signature_scoring_func.parameters
+            and "pos_label" not in scorer_kwargs
+        ):
+            if pos_label is None:
+                # Since the provided `pos_label` is the default, we need to
+                # use the default value of the scoring function that can be either
+                # `None` or `1`.
+                scorer_kwargs["pos_label"] = signature_scoring_func.parameters[
+                    "pos_label"
+                ].default
+            else:
+                scorer_kwargs["pos_label"] = pos_label
+        # transform a binary metric into a curve metric for all possible decision
+        # thresholds
+        instance = cls(
+            score_func=scorer._score_func,
+            sign=scorer._sign,
+            response_method=response_method,
+            n_thresholds=n_thresholds,
+            kwargs=scorer_kwargs,
+        )
+        # transfer the metadata request
+        instance._metadata_request = scorer._get_metadata_request()
+        return instance
+
+    def _score(self, method_caller, estimator, X, y_true, **kwargs):
+        """Evaluate predicted target values for X relative to y_true.
+
+        Parameters
+        ----------
+        estimator : object
+            Trained estimator to use for scoring.
+
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Test data that will be fed to estimator.predict.
+
+        y_true : array-like of shape (n_samples,)
+            Gold standard target values for X.
+
+        **kwargs : dict
+            Other parameters passed to the scorer. Refer to
+            :func:`set_score_request` for more details.
+
+        Returns
+        -------
+        score : float
+            Score function applied to prediction of estimator on X.
+        """
+        pos_label = self._get_pos_label()
+        y_score = method_caller(
+            estimator, self._response_method, X, pos_label=pos_label
+        )
+
+        scoring_kwargs = {**self._kwargs, **kwargs}
+        if isinstance(self._n_thresholds, Integral):
+            potential_thresholds = np.linspace(
+                np.min(y_score), np.max(y_score), self._n_thresholds
+            )
+        else:
+            potential_thresholds = np.array(self._n_thresholds, copy=False)
+        score_thresholds = [
+            self._sign
+            * self._score_func(
+                y_true,
+                _threshold_scores_to_class_labels(
+                    y_score, th, estimator.classes_, self._get_pos_label()
+                ),
+                **scoring_kwargs,
+            )
+            for th in potential_thresholds
+        ]
+        return potential_thresholds, np.array(score_thresholds)
+
+
+def _estimator_has(attr):
```

Comment:
isn't this in utils now?

Reply:
Nop but we should make a refactoring at some point. This could avoid some bugs that we already solved.